### PR TITLE
Move upgrading guide for keycloak-client libraries to the client docu…

### LINKED ISSUE #93: Figure how to do release notes & upgrading guide for keycloak-client
### Description

Do we want release notes & upgrading guide for keycloak-client? Few considerations for the discussion:

- Can we merge both release notes & upgrading guide in the single guide? There will be probably very little in the upgrading guide as we try to have it backwards compatible. For the release notes, we can possibly sometimes add like "This client version is fully compatible with Keycloak server 26.1.0" to notify people that particular keycloak-client version contains endpoints, which are applicable for keycloak server 26.

- Where to publish it? Maybe for the community, it can be separate section under "Securing applications" inside https://www.keycloak.org/guides . This means that we can keep the release notes & upgrading guide alongside other keycloak-client documentation and it will be properly re-published on the keycloak web page during keycloak-client release.

- For the product, not sure yet (probably dedicated discussion)

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by mposolda**
…mentation

closes #93

Moving this section https://www.keycloak.org/docs/latest/upgrading/index.html#upgrading-the-keycloak-client-libraries from the server documentation to the client repository documentation as it is related to clients.

Once the keycloak-client 26.0.3 is released and this is published on the web, there would be follow-up task for removing this part from the server documentation - https://github.com/keycloak/keycloak/issues/34949

### CODE REVIEW

File: docs/guides/securing-apps/upgrading.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+<@tmpl.guide
+title="Upgrading the {project_name} Client Libraries"
+priority=540
+summary="How to upgrade the {project_name} Client Libraries">
+
+The client libraries are those artifacts:
+
+* Java admin client - Maven artifact `org.keycloak:keycloak-admin-client`
+* Java authorization client - Maven artifact `org.keycloak:keycloak-authz-client`
+* Java policy enforcer - Maven artifact `org.keycloak:keycloak-policy-enforcer`
```
Comment (pskopek) [inline docs/guides/securing-apps/upgrading.adoc:14]: Shouldn't we mention `org.keycloak:keycloak-client-common-synced` as a library containing data representation classes and necessary common classes?

File: docs/guides/attributes.adoc [hunk 2]
```diff
@@ -8,5 +8,7 @@
 :developerguide_name: Server Developer Guide
 :developerguide_name_short: Server Developer
 :developerguide_link: {project_doc_base_url}/server_development/
+:project_product: false
```
Comment (jonkoops) [inline docs/guides/attributes.adoc:11]: @drichtarik FYI

**Review by jonkoops — APPROVED** [review]

File: docs/guides/securing-apps/upgrading.adoc (hunk 1 above)
Comment (mposolda) [reply docs/guides/securing-apps/upgrading.adoc:14]: Thanks for pointing this! Added the additional line with this library.

**Review by pskopek — APPROVED** [review]
LGTM!

**Pull request merged**